pkg/referenceutil/referenceutil.go

```diff
@@ -32,6 +32,20 @@ type Reference interface {
 	String() string
 }
 
+// ParseAnyReference parses the passed reference as IPFS, CID, or a classic reference.
+// Unlike ParseAny, it is not limited to the DockerRef limitations (being either tagged or digested)
+// and should be used instead.
+func ParseAnyReference(rawRef string) (Reference, error) {
+	if scheme, ref, err := ParseIPFSRefWithScheme(rawRef); err == nil {
+		return Reference(stringRef{scheme: scheme, s: ref}), nil
+	}
+	if c, err := cid.Decode(rawRef); err == nil {
+		return c, nil
+	}
+	return refdocker.ParseAnyReference(rawRef)
+}
+
+// Deprecated
```

Comment:
Propose that we stop using ParseDockerRef in favor of ParseAnyReference. The former has some nasty side effects by dropping the tag in case there is a digest AFAIK.

Reply:
I don't think it doesn't need to be `Deprecated` as it's still used in the codebase. But I agree with ParseAnyReference should be used for necessary places.

Reply:
Let me remove the deprecated tag.

Thanks a lot for your review @ktock 
Really appreciated!